libslide/src/partial_evaluator/rules.rs

```diff
@@ -0,0 +1,83 @@
+use crate::grammar::{Expr, Stmt};
+use crate::{parse, scan, ScannerOptions};
+
+use core::fmt;
+
+// TODO: will uncomment in a future iteration when we support removing rules.
+// pub enum RuleName {
+//     VariablePlusZero = 0,
+// }
+
+static DEFAULT_RULESET: &[&'static str] = &[
+    "$a + 0 -> $a", // VariablePlusZero
+];
+
+pub struct RuleSet {
+    rules: Vec<String>,
+}
+
+impl Default for RuleSet {
+    fn default() -> Self {
+        Self {
+            rules: DEFAULT_RULESET.iter().map(|s| s.to_string()).collect(),
+        }
+    }
+}
+
+impl RuleSet {
+    /// Creates a list of `BuiltRule`s from the rule set.
+    pub fn build(&self) -> Vec<BuiltRule> {
+        self.rules.iter().map(BuiltRule::from).collect()
+    }
+}
+
+pub struct BuiltRule {
+    from: Expr,
+    to: Expr,
+}
+
+impl From<&String> for BuiltRule {
+    /// Converts a string representation of a rule to a `BuiltRule`.
+    /// A rule must be expressed in the string form
+    ///   <expr> -> <expr>
+    /// Where <expr> is an expression.
+    fn from(rule: &String) -> Self {
+        fn is_var_char(c: char) -> bool {
+            return c.is_alphabetic() || c == '$' || c == '#' || c == '_';
```

Comment:
why do we have $ # and _ here?

Reply:
See the grammar in the linked issue. Later we will use # and _ for other variables.